### PR TITLE
Check cluster ufs version in getClusterConf

### LINKED ISSUE #18221: Writing to ozone failed without setting alluxio.underfs.version
**Alluxio Version:**
2.x

**Describe the bug**
Writing to ozone failed without setting alluxio.underfs.version.
```
WARN  ExtensionFactoryRegistry - No factory implementation supports the path o3fs://xxx.xxx/
WARN  UnderFileSystemFactoryRegistry - Versions [xxx] are supported for path o3fs://xxx.xxx/ but you have configured version: xxx
```
```
java.lang.IllegalArgumentException: No Under File System Factory found for: o3fs://xxx.xxx/
        at alluxio.underfs.UnderFileSystem$Factory.create(UnderFileSystem.java:95)
        at alluxio.underfs.AbstractUfsManager.getOrAdd(AbstractUfsManager.java:132)
        at alluxio.underfs.AbstractUfsManager.lambda$addMount$0(AbstractUfsManager.java:183)
        at alluxio.underfs.UfsManager$UfsClient.acquireUfsResource(UfsManager.java:61)
        at alluxio.worker.grpc.UfsFileWriteHandler.createUfsFile(UfsFileWriteHandler.java:155)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:139)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:49)
        at alluxio.worker.grpc.AbstractWriteHandler.writeData(AbstractWriteHandler.java:281)
        at alluxio.worker.grpc.AbstractWriteHandler.lambda$writeDataMessage$1(AbstractWriteHandler.java:179)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at alluxio.worker.grpc.GrpcExecutors$ImpersonateThreadPoolExecutor.lambda$execute$0(GrpcExecutors.java:159)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**
1. Start a cluster without setting alluxio.underfs.version;
2. Add a mount point for an ozone bucket without setting alluxio.underfs.version;
3. Upload a file to the ozone bucket.

**Expected behavior**
Can write files.

**Are you planning to fix it**
Yes.

### PR DESCRIPTION
**Pull request by Haoning-Sun**
### What changes are proposed in this pull request?

Check cluster ufs version in getClusterConf.

### Why are the changes needed?

Fix #18221.

### CODE REVIEW

**Comment by jja725** [comment]
hey @jiacheliu3 , do you mind taking a look at this change? The change looks good to me but not sure that's the correct fix.

**Comment by yyongycy** [comment]
@jja725 what is your opinion?

**Review by jja725 — APPROVED** [review]

File: core/common/src/main/java/alluxio/conf/Configuration.java [hunk 1]
```diff
@@ -493,6 +494,11 @@ public static InstancedConfiguration getClusterConf(GetConfigurationPResponse re
           scope, clientVersion, clusterVersion);
       clusterProps.remove(PropertyKey.VERSION);
     }
+    // Check ufs version. Avoid adding it to user properties if the two versions are the same.
+    String clusterUfsVersion = clusterProps.get(PropertyKey.UNDERFS_VERSION).toString();
```
Comment (yyongycy) [inline core/common/src/main/java/alluxio/conf/Configuration.java:498]: could you please clarify why the version caused problem?

**Comment by Haoning-Sun** [comment]
@yyongycy 
The #18221 error is as follows
```
java.lang.IllegalArgumentException: No Under File System Factory found for: o3fs://xxx.xxx/
        at alluxio.underfs.UnderFileSystem$Factory.create(UnderFileSystem.java:95)
        at alluxio.underfs.AbstractUfsManager.getOrAdd(AbstractUfsManager.java:132)
        at alluxio.underfs.AbstractUfsManager.lambda$addMount$0(AbstractUfsManager.java:183)
        at alluxio.underfs.UfsManager$UfsClient.acquireUfsResource(UfsManager.java:61)
        at alluxio.worker.grpc.UfsFileWriteHandler.createUfsFile(UfsFileWriteHandler.java:155)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:139)
        at alluxio.worker.grpc.UfsFileWriteHandler.writeBuf(UfsFileWriteHandler.java:49)
        at alluxio.worker.grpc.AbstractWriteHandler.writeData(AbstractWriteHandler.java:281)
        at alluxio.worker.grpc.AbstractWriteHandler.lambda$writeDataMessage$1(AbstractWriteHandler.java:179)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at alluxio.worker.grpc.GrpcExecutors$ImpersonateThreadPoolExecutor.lambda$execute$0(GrpcExecutors.java:159)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

The ufs info is obtained by `mMasterClient.getUfsInfo`. If `underfs.version` is set during the mount operation, the value in the configuration will be overwritten in `createMountSpecificConf`, otherwise the value in the current configuration will be used when creating ufs client.
image

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by yyongycy** [comment]
alluxio-bot, merge this please

**Pull request merged**